cmd/strelaysrv/listener.go

```diff
@@ -82,7 +83,7 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config, token strin
 	}
 
 	state := conn.ConnectionState()
-	if debug && state.NegotiatedProtocol != protocol.ProtocolName {
+	if debug && !slices.Contains(config.NextProtos, state.NegotiatedProtocol) {
```

Comment:
I am speculating, but given the message printed below, I interpret this as a defensive condition to find any kind of misconfig/bug, leading to another protocol than the desired to be used. So specifying the desired protocol is the intent - otherwise a misconfig of `config` wouldn't be caught.